### ISSUE TITLE
Feature: Exclude entries from generated treemap

### ISSUE THREAD

**Issue by joscha** [webpack-contrib/webpack-bundle-analyzer#71]
#### Issue description

We have a project where we generate multiple entry points at once, but we would like the stats to be reported independently (per entry point).

Pretty much a whitelist (or blacklist) option for the entry points that should be included int he analyzer output.

I tried using the `statsOptions`, but after reading through some of the source I think this has no effect on the generated visuals. The change would probably need to live somewhere here: https://github.com/th0r/webpack-bundle-analyzer/blob/master/src/analyzer.js#L29-L35

#### Technical info
* Webpack Bundle Analyzer version: 2.6.0
* Webpack version: 2.4.1
* Node.js version: 7.10
* npm/yarn version: yarn 0.23.2
* OS: Sierra

#### Debug info
How do you use this module? As CLI utility or as plugin?

as a plugin via:

```
new BundleAnalyzerPlugin({
        analyzerMode: 'static',
        defaultSizes: 'stat',
        reportFilename: 'stats.html',
        openAnalyzer: false,
        statsOptions: {
          exclude: [/editor/]  // <- here is my attempt to hide a specific chunk
        },
      })
```

Happy to open a pull request to make this work.

**Comment by valscion** [comment]
Sounds like a legit use case :+1:

I'm not really sure how I'd go about testing this feature, though. I'm not really sure whether there's a good slot for this in the existing testing infra, maybe th0r knows better.

**Comment by joscha** [comment]
^ @th0r any thoughts on this?

**Comment by th0r** [comment]
@joscha Let me confirm that I understood you correctly. You are saying that you have multiple entry points, for example, `app` and `vendor` and two resulting bindles: `app.bundle.js` and `vendor.bundle.js` respectively. And you want stats for them to be generated separately into two different html files, right?

**Comment by joscha** [comment]
@th0r yes, that is correct. I want to be able to hide all other unrelated bundles and/or whitelist a specific one, so I can make sure that only ever one shows in the generated stats file.

**Comment by th0r** [comment]
> only ever one shows in the generated stats file

@joscha Wait. Do you want to exclude information from the **stats** file (`stats.json` that's generated when `generateStatsFile` option is `true`) or the **report** file (`report.html` with the treemap)?

**Comment by joscha** [comment]
> the report file (report.html with the treemap)

^ this one!

**Comment by th0r** [comment]
Ok. What if there would be a list of all chunks with checkboxes in the left panel under `Treemap sizes` so you'll be able to filter out all unnecessary ones? Would that work for you?

**Comment by joscha** [comment]
@th0r that would be cool as well, yes!

**Comment by th0r** [comment]
`v2.7.0` shows the checkbox list of all chunks in the sidebar.

**Comment by joscha** [comment]
@th0r sorry to reopen this issue, but the workaround to use the checkbox doesn't work any longer. Background is, that we generate a language file for each language we support (it's an entry chunk), making the stats look like this:

![webpack_bundle_analyzer](https://user-images.githubusercontent.com/188038/39345299-06bdf10e-4a2b-11e8-916b-28a02bd36ebe.png)

I tried to remove the chunks from the stats output via:
```
        statsOptions: {
          excludeModules: /lang/
        }
```

but it doesn't seem to work as I expect it to. Any chance for an option that allows me to exclude entries after all?

**Comment by th0r** [comment]
> but the workaround to use the checkbox doesn't work any longer

May I ask why? You can use `All` checkbox to hide all chunks and check only those you need.

**Comment by joscha** [comment]
friendly ping @th0r ?

**Comment by th0r** [comment]
@joscha I'll take a look when I have time. Or PRs are always welcome 😉

**Comment by valscion** [comment]
If `statsOptions.excludeModules` is a thing, and we could act upon it, then that would be nice. We wouldn't need to add extra configuration to `webpack-bundle-analyzer` in that case :)

**Comment by th0r** [comment]
Released `v2.12.0` with `excludeAssets` plugin option and `--exclude` CLI option.

**Comment by joscha** [comment]
works a charm @th0r, thank you!
